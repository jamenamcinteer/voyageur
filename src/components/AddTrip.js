import React from "react";
import Header from "./Navigation/Header";
import TripForm from "./Forms/TripForm";
import { PageSection } from "./StyledComponents/Layout";
import { connect } from "react-redux";

const AddTrip = props => {
  return (
    <React.Fragment>
      <Header
        title="Add a Trip"
        theme={props.theme}
        backTo="/"
        auth={props.auth}
      />
      <PageSection>
        <TripForm theme={props.theme} history={props.history} />
      </PageSection>
    </React.Fragment>
  );
};

const mapStateToProps = (state, props) => {
  return {
    auth: state.auth
  };
};

export default connect(mapStateToProps)(AddTrip);
