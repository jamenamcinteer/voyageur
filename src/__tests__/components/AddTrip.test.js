import React from "react";
import { render, cleanup } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import tripsReducer, { tripsReducerDefaultState } from "../../reducers/trips";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import "react-dates/initialize";
import AddTrip from "../../components/AddTrip";
import { auth } from "../../testUtils/fixtures/mockStoreData";

function renderWithReduxAndRouter(
  ui,
  {
    initialState,
    store = createStore(tripsReducer, tripsReducerDefaultState),
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{ui}</Router>
      </Provider>
    ),
    store,
    history
  };
}

afterEach(cleanup);

jest.mock("uniqid", () => {
  let value = 1;
  return () => value++;
});

test("can render with redux with custom store", () => {
  const store = createStore(() => ({
    auth
  }));
  const { asFragment } = renderWithReduxAndRouter(<AddTrip />, {
    store
  });
  expect(asFragment()).toMatchSnapshot();
});
