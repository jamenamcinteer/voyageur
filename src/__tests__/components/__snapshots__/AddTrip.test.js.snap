// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with redux with custom store 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-bdVaJa hZqfbl"
    >
      <a
        href="/"
      >
        <svg
          style="fill: #4b5c6b; width: 20px; height: 20px; float: left; margin-top: 2px;"
          version="1.1"
          viewBox="0 0 199.404 199.404"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <polygon
              points="135.412,0 35.709,99.702 135.412,199.404 163.695,171.119 92.277,99.702 163.695,28.285 	"
            />
          </g>
        </svg>
      </a>
      <div
        class="sc-bwzfXH ebZROL"
      >
        Add a Trip
      </div>
      <div
        class="sc-htpNat KtFsv"
      >
        <img
          alt=""
          class="sc-bxivhb bfhPyW"
          src="https://picsum.photos/id/947/45/45"
        />
      </div>
    </div>
    <div>
      <div
        class="sc-fjdhpX cEgVjZ"
      >
        <div
          class="sc-bZQynM blxRiI"
        >
          <label
            class="sc-gzVnrw sohPS"
            for="1"
          >
            Destination
          </label>
          <input
            class="sc-htoDjs hNiZih"
            id="1"
            type="text"
            value=""
          />
        </div>
        <div />
        <div
          class="DateRangePicker DateRangePicker_1"
        >
          <div>
            <div
              class="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2"
            >
              <div
                class="DateInput DateInput_1"
              >
                <input
                  aria-describedby="DateInput__screen-reader-message-startDate"
                  aria-label="Start Date"
                  autocomplete="off"
                  class="DateInput_input DateInput_input_1"
                  id="startDate"
                  name="startDate"
                  placeholder="Start Date"
                  type="text"
                  value=""
                />
                <p
                  class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                  id="DateInput__screen-reader-message-startDate"
                >
                  Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                </p>
              </div>
              <div
                aria-hidden="true"
                class="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
                role="presentation"
              >
                <svg
                  class="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
                  focusable="false"
                  viewBox="0 0 1000 1000"
                >
                  <path
                    d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
                  />
                </svg>
              </div>
              <div
                class="DateInput DateInput_1"
              >
                <input
                  aria-describedby="DateInput__screen-reader-message-endDate"
                  aria-label="End Date"
                  autocomplete="off"
                  class="DateInput_input DateInput_input_1"
                  id="endDate"
                  name="endDate"
                  placeholder="End Date"
                  type="text"
                  value=""
                />
                <p
                  class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                  id="DateInput__screen-reader-message-endDate"
                >
                  Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div />
        <div>
          <div
            class="sc-chPdSV iCLCng"
          />
        </div>
        <div
          class="sc-kAzzGY LGZmy"
        >
          <div
            class="sc-cSHVUG eJVCam"
          >
            <a
              backgroundcolor="transparent"
              border="0"
              buttontype="link"
              buttonwidth="auto"
              class="sc-gqjmRU cVvmHR"
              href="/"
              style="padding: 10px 0px;"
            >
              <span
                class="sc-VigVT hBprvu"
              >
                Cancel
              </span>
            </a>
            <button
              class="sc-iwsKbI NtbZn"
            >
              <span
                class="sc-gZMcBi cnsAlJ"
                color="#fff"
              >
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
